sortinghat/core/jobs.py

```diff
@@ -213,6 +213,126 @@ def affiliate(ctx, uuids=None):
     return job_result
 
 
+@django_rq.job
+def unify(ctx, source_uuids, target_uuids, criteria):
+    """Unify a set of individuals by merging them using matching recommendations.
+
+    This function automates the identities unify process obtaining
+    a list of recommendations where matching individuals can be merged.
+    After that, matching individuals are merged.
+    This job returns a list with the individuals which have been merged
+    and the errors generated during this process.
+
+    Individuals both for `source_uuids` and `target_uuids` are defined by
+    any of their valid keys or UUIDs. When the parameter `target_uuids` is
+    empty, the job will take all the individuals stored in the registry,
+    so matches will be found comparing the identities from the individuals
+    in `source_uuids` against all the identities on the registry.
+
+    :param ctx: context where this job is run
+    :param source_uuids: list of individuals identifiers to look matches for
+    :param target_uuids: list of individuals identifiers where to look for matches
+    :param criteria: list of fields which the unify will be based on
+        (`email`, `name` and/or `username`)
+
+    :returns: a list with which individuals were merged
+        and the errors found running the job
+    """
+    def _group_recommendations(recs):
+        """Calculate unique sets of identities from matching recommendations.
+
+        For instance, given a list of matching groups like
+        A = {A, B}; B = {B,A,C}, C = {C,} and D = {D,} the output
+        for keys A, B and C will be the group {A, B, C}. As D has no matches,
+        it won't be included in any group and it won't be returned.
+
+        :param recs: recommendations of matching identities
+
+        :returns: a list including unique groups of matches
+        """
+        groups = []
+        for group_key in recs:
+            g_uuids = recs[group_key]
+            if group_key not in g_uuids:
+                g_uuids.append(group_key)
+            g_uuids = sorted(g_uuids)
+            if (len(g_uuids) > 1) and (g_uuids not in groups):
```

Comment:
Why do we need to check if the `g_uuids` are in `groups`?

Reply:
Please check my answer to this comment: https://github.com/chaoss/grimoirelab-sortinghat/pull/312#discussion_r466906138.